src/main/java/com/cloudbees/hudson/plugins/folder/computed/DeletingChildrenException.java

```diff
@@ -0,0 +1,20 @@
+package com.cloudbees.hudson.plugins.folder.computed;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DeletingChildrenException extends RuntimeException {
```

Comment:
This should be an `IOException` with the cause set to the first exception in the list.
That way it would be much easier finding the reason for the error as the stack trace in the log would show the first error as well, and the likelihood of problems deleting are the same is very high.